external_search.py

```diff
@@ -361,11 +317,23 @@ def setup_index(self, new_index=None, **index_settings):
             self.indices.delete(index_name)
 
         self.log.info("Creating index %s", index_name)
-        body = ExternalSearchIndexVersions.latest_body()
+        body = self.mapping.body()
         body.setdefault('settings', {}).update(index_settings)
-        index = self.indices.create(
-            index=index_name, body=body
-        )
+        index = self.indices.create(index=index_name, body=body)
+
+    def set_stored_scripts(self):
```

Comment:
Since the stored scripts go along with the mapping, they ought to be defined in the mapping, not in ExternalSearchIndex. But ExternalSearchIndex is in charge of storing them on the server, just as it's in charge of sending the mapping document to the server.